### ISSUE TITLE
[bug]: InvokeAI 2.2.5 localhost server not connecting 'This site can’t be reached' on Windows

### ISSUE THREAD

**Issue by shashankitsoft** [invoke-ai/InvokeAI#2274]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

Windows

### GPU

cuda

### VRAM

see screenshot

### What happened?

I have installed InvokeAI 2.2.5 using automated windows 11 installer method using bat file. I have python version 3.9.13.   I have just chosen D:\InvokeAI as the installation path instead of C drive so my installed path is D:\InvokeAI\invokeai\ and the outputs folder path is D:\InvokeAI\invokeai\outputs\.  Installation done properly.
But when I run invoke.bat to start Web broswer based UI (option 2) it just say HaveFun in last with localhost url.
But when I open http://localhost:9090 in chrome browser it gives error "This site can’t be reached. localhost refused to connect."

I tried installing multiple times, sometimes skipping models (s) download, sometime with recommend models (r) and sometime with all models (a) download, but every ways have same issue.  Hugging face registered, token etc added when asked, but not able to open invokeai in browser. 

My windows is updated, latest drivers, Intel UHD graphics and Nvidia GeForce GTX 1650 Ti card, VS C++ prerequisite 2015-22 also installed. Processor: Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz   2.59 GHz, RAM: 16GB. Restarted laptop several times. Everything done whatever mentioned in guide. 
What can make it work? 

### Screenshots

![Device Manager](https://user-images.githubusercontent.com/18620683/211192324-df10c615-015f-48ed-bdcd-02f13e254c88.png)
![Settings-Display](https://user-images.githubusercontent.com/18620683/211192328-0caf3f7e-667a-49f0-a6f0-18440e3a3090.png)
![InvokeAI-start](https://user-images.githubusercontent.com/18620683/211192337-88599f49-b910-4ee5-9023-51a47e8fce65.png)
![localhost-site-error](https://user-images.githubusercontent.com/18620683/211192344-fac19626-fa70-4f5b-bd96-855856ef1d2b.png)
![invokeai-installed-folder](https://user-images.githubusercontent.com/18620683/211192622-f9a28b28-f044-41dc-af39-de11d70cc4fc.png)

**Comment by ws132** [comment]
It's because you don't have a usable mod installed.
The easy way is to register a huggingface account and go to the following [link](https://huggingface.co/settings/tokens)
Get the tokens, and follow the script prompts to continue.

**Comment by shashankitsoft** [comment]
@ws132 As mentioned in my description, I have already registered HuggingFace account, got token and added in script prompt whenever asked.
There was no error shown in installation or running. The only thing is I am unable to access the web GUI in browser on localhost:9090 or 127.0.0.1:9090  . The last Have Fun! text it seems doesn't actually starting the local server.

**Comment by ws132** [comment]
> @ws132 As mentioned in my description, I have already registered HuggingFace account, got token and added in script prompt whenever asked. There was no error shown in installation or running. The only thing is I am unable to access the web GUI in browser on localhost:9090 or 127.0.0.1:9090 . The last Have Fun! text it seems doesn't actually starting the local server.

Does the bat script exit automatically after running?
You can try to run it on the terminal to see if there is any error message, or you can check whether there is a stable-diffusion-v1 folder in the models\ldm file.
In the stable-diffusion-v1 folder, there should be model files in ckpt format.

**Comment by ws132** [comment]
Also, I noticed that your python version is 3.9, you can try version 3.10.
I'm using 3.10.9 and it works fine.

**Comment by shashankitsoft** [comment]
@ws132 Yes bat script exits automatically. Everytime I run invoke.bat ans press option 2 (web gui) it asks to download models, I press r (recommended), a (all), or s (skip), in any case it checks model exists, then Have Fun and exit to command prompt.
Yes, I am already running in terminal. No error message. As I have attached screenshots in the issue detail.
Yes there is stable-diffusion-v1 folder in the models\ldm, and also the ckpt files inside. Attaching modals folder screenshot:

![models](https://user-images.githubusercontent.com/18620683/211351363-1010fcdf-7d45-4de6-933c-2a785195c605.png)
![models-ldm-stable-diffusion-v1](https://user-images.githubusercontent.com/18620683/211351412-963dfaf8-2fa4-474f-886e-f902dd747cde.png)

3.9 is the recommended version in invokeAI docs so I downgraded my python from 3.11.x to 3.9.x .

**Comment by AryLFC** [comment]
> ### Is there an existing issue for this?
> * [x]  I have searched the existing issues
> 
> ### OS
> Windows
> 
> ### GPU
> cuda
> 
> ### VRAM
> see screenshot
> 
> ### What happened?
> I have installed InvokeAI 2.2.5 using automated windows 11 installer method using bat file. I have python version 3.9.13. I have just chosen D:\InvokeAI as the installation path instead of C drive so my installed path is D:\InvokeAI\invokeai\ and the outputs folder path is D:\InvokeAI\invokeai\outputs. Installation done properly. But when I run invoke.bat to start Web broswer based UI (option 2) it just say HaveFun in last with localhost url. But when I open http://localhost:9090 in chrome browser it gives error "This site can’t be reached. localhost refused to connect."
> 
> I tried installing multiple times, sometimes skipping models (s) download, sometime with recommend models (r) and sometime with all models (a) download, but every ways have same issue. Hugging face registered, token etc added when asked, but not able to open invokeai in browser.
> 
> My windows is updated, latest drivers, Intel UHD graphics and Nvidia GeForce GTX 1650 Ti card, VS C++ prerequisite 2015-22 also installed. Processor: Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz 2.59 GHz, RAM: 16GB. Restarted laptop several times. Everything done whatever mentioned in guide. What can make it work?
> 
> ### Screenshots
> ![Device Manager](https://user-images.githubusercontent.com/18620683/211192324-df10c615-015f-48ed-bdcd-02f13e254c88.png) ![Settings-Display](https://user-images.githubusercontent.com/18620683/211192328-0caf3f7e-667a-49f0-a6f0-18440e3a3090.png) ![InvokeAI-start](https://user-images.githubusercontent.com/18620683/211192337-88599f49-b910-4ee5-9023-51a47e8fce65.png) ![localhost-site-error](https://user-images.githubusercontent.com/18620683/211192344-fac19626-fa70-4f5b-bd96-855856ef1d2b.png) ![invokeai-installed-folder](https://user-images.githubusercontent.com/18620683/211192622-f9a28b28-f044-41dc-af39-de11d70cc4fc.png)
> 
> ### Additional context
> _No response_

I have EXACTLY the same problem like this...so I'm watching this discussion quite curious 😃

**Comment by ws132** [comment]
Confirm that the settings in the invokeai.init file are correct. If necessary, delete this file and re-run invoke.bat
If it still doesn't work, try to start with parameters, for example, -lowvram --precision full --no-half

**Comment by ws132** [comment]
> 
You need to open the invokeai.init file and add startup parameters in it.
I have no way to determine whether this is due to SD, or a problem with invokeAI.🤔
Have you tried using other webui?
For example, [stable-diffusion-webui](https://github.com/AUTOMATIC1111/stable-diffusion-webui)

**Comment by irebix** [comment]
Same question, waiting for an answer...

**Comment by AryLFC** [comment]
> Have you tried using other webui?

Yes, I have...It was Midjourney and so far quite 'perfect' AI generator for me since I haven't tried Invoke. That's why I feel quite enthusiastically want to try this Invoke AI. But let me try Stable Diffusion WebUI then...

**Comment by 6630507** [comment]
Try turning of your firewall and then connecting.  WebUI might use a different port number than Invoke.

**Comment by 2zsoldi2** [comment]
Same problem here. Firewall settings done, but still doesnt wanna connect at all.

**Comment by ws132** [comment]
I think of a question, do you have enough disk space to use?
SD seems to use a lot of cache space on the c drive, if there is not enough space, it may not be able to run.

**Comment by 2zsoldi2** [comment]
This cant be the issue. Have plenty of space. Automatic 1111 runs smooth.

**Comment by AryLFC** [comment]
> I think of a question, do you have enough disk space to use? SD seems to use a lot of cache space on the c drive, if there is not enough space, it may not be able to run.

My SD is running very well...but still have no clue with Invoke...

**Comment by shashankitsoft** [comment]
> Confirm that the settings in the invokeai.init file are correct. If necessary, delete this file and re-run invoke.bat If it still doesn't work, try to start with parameters, for example, -lowvram --precision full --no-half

@ws132 I deleted invokeai.init file and regenrated. Then added parameters in it as you suggested, but it gives error in running with those parameters. For ex.  `error: argument --precision: invalid choice: 'full' (choose from 'auto', 'float32', 'autocast', 'float16')`. 
See attached screenshot.
Also neither --lowvram nor --no-half is mentioned in command line as valid options. When I remove precision argument it gives,  `error: unrecognized arguments: --lowvram --no-half`

Please suggest a valid argument which I can run and test.

Yes I have sufficient disk space in C as well D drive. Btw, I am installing in **D drive** not C.

![invokeai init](https://user-images.githubusercontent.com/18620683/212391705-feae5983-4274-47b0-a16e-9167164b6066.png)

**Comment by ws132** [comment]
> 

Sorry, parameters should be used in invoke.init
```
--full_precision
```
Other parameters I did not see supported options in the documentation.
If it still fails to start, please try to see if it can be started in command-line mode.
In addition, I also read in the documentation that developers do not recommend the use of 10xx series graphics cards and 1650 and 1660 series graphics cards.
But I'm using a 1060 6GB and it's working fine, so it should not be a problem.

**Comment by shashankitsoft** [comment]
@ws132 Tried --full_precision , but it says this is deprecated and suggest to use --precision float32 instead. That also not worked.
Although tried running again several times with no additional flags, I found below error few times:
` model stable-diffusion-1.5 could not be loaded: CUDA out of memory. Tried to allocate 20.00 MiB (GPU 0; 4.00 GiB total capacity; 3.39 GiB already allocated; 0 bytes free; 3.45 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF`
It seems the error is in loading stable-diffusion-1.5 model.
See attached screenshot:
![invokeai-memory](https://user-images.githubusercontent.com/18620683/212461737-e4ef9477-4a0b-467d-bd04-d2ca472da7fa.png)


Here is the screenshot from my task manager GPU1 : Nvidia (4GB dedicated gpu, 8 GB shared gpu) and GPU0:  Intel UHD Graphics (8gb gpu). Currently being used around 0, so all available. Yes 1650 and 1660 series they say not recommended, but it doesn't mean it cannot run (may be with not full performance, but it should start).

![gpu1](https://user-images.githubusercontent.com/18620683/212461868-befa0f81-4cfd-4725-80e3-24813b90ce33.png)

![gpu0](https://user-images.githubusercontent.com/18620683/212461877-fd7945b2-ccd8-4e87-8ac5-d3cc43111e24.png)

Searching about that 'CUDA out of memory' error, I found this thread https://github.com/CompVis/stable-diffusion/issues/39 which suggests:
**Way1:**  to run
`set PYTORCH_CUDA_ALLOC_CONF=garbage_collection_threshold:0.6,max_split_size_mb:128`
(which I did in command line) and then to run with `--n_samples 1`  (which I don't know how, as invoke.bat and invokeai.init not supporting this argument). Can I use --n_samples parameter somewhere in invokeai?

**Way2:** It also suggests to use https://github.com/basujindal/stable-diffusion which is Optimized Stable Diffusion modified to run on lower GPU VRAM (as good solution), but dont know how to use that in InvokeAI. Can anyone help?

**Way3:** The https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Troubleshooting also suggests to use `--lowvram --opt-split-attention` as sure solving solution, but how we can use those argument in invokeai is not clear. **InvokeAI should also supports such important arguments.**

**Way4:** I have 4gb dedicated + 8 gb shared GPU (as in screenshot). Is there way to make invokeAI not consider using only the 4gb and use both instead. Or some PyTorch configuration?

Any of the above 4 ways I guess can make it work if rightly done. Not found anywhere documented for InvokeAI, that's sad.

**Comment by ws132** [comment]
Maybe you can try the command line mode, refer to [here](https://invoke-ai.github.io/InvokeAI/features/CLI/).
As for other methods, I didn't find relevant information, maybe it would be a good idea to talk to the developers directly.
Perhaps also using a web platform such as [Google Colab](https://colab.research.google.com/).

**Comment by shashankitsoft** [comment]
Yes tried to open command line with choosing option 1, but that also failing to load SD1.5 and giving CUDA out of memory error.

Yeah, waiting for some InvokeAI developer to look over this issue and suggest something.

**Comment by shashankitsoft** [comment]
For the note: I have installed Python 3.10.9 to test, but InvokeAI giving same problem with this version too.

To check if the problem is with Stable Diffusion model, I installed the https://github.com/AUTOMATIC1111/stable-diffusion-webui **Automatic SD webui** with **SD model 1.4** as well as **1.5** , and both worked in a go, without needing any arguments. Very easy installation and running webui and generating images. :)
Even to push harder on my system I ran this webui by adding **Stable Diffusion 2.1 (latest) model** (512 & 768 both) and even that worked flawless (just needed to add argument only in this case). And great thing is their arguments works perfect to run in browser.
I am currently having all three 1.4, 1.5 and 2.1 SD models and able to switch in browser easily and generating images with Automatic SD webui.

InvokeAI should also optimize this thing like that. Hoping some positive update from InvokeAI developers. Wanna see InvokeAI running on my browser as well.
